src/model.ts

```diff
@@ -5,30 +5,147 @@ import {base, BaseType} from "./base";
 
 import {InternalStorage, UserType} from "./user";
 
-// Probably this is something that dopri should export for us, we
-// could also use its types for the rhs and output members below.
+/** Interpolated solution to the system of differential equations
+ *
+ *  @param t The time to look up the solution at
+ */
 export type Solution = (t: number) => number[];
 
+/**
+ * Constructor for an {@link OdinModel}
+ *
+ * @param base The singleton object {@link base}
+ *
+ * @param pars Parameters to pass to the model
+ *
+ * @param unusedUserAction String, describing the action to take if
+ * there are unknown values in `pars` - possible values are "error",
+ * "ignore", "warning" and "message"
+ */
 export type OdinModelConstructable =
-    new(base: BaseType, pars: UserType, unknownAction: string) => OdinModel;
-
-interface OdinModelODE {
-    setUser(pars: UserType, unknownAction: string): void;
+    new(base: BaseType, pars: UserType, unusedUserAction: string) => OdinModel;
+
+/**
+ * The interface that odin ordinary differential equation (ODE) models
+ * will match, though they will be generated in plain JavaScript and
+ * not TypeScript.
+ */
+export interface OdinModelODE {
```

Comment:
Are OdinModelODE and OdinModelDDE actually identical (or nearly?) - could this just be one interface, or extended from a common interface? Should have asked this in the last PR really...

Reply:
They differ in their rhs and output methods. We could use a base method and extend both (probably in a new PR though?)

Reply:
actually this is not bad to implement, will do it now